src/panels/lovelace/entity-rows/hui-humidifier-entity-row.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  PropertyValues,
+  TemplateResult,
+} from "lit-element";
+import "../../../components/entity/ha-entity-humidifier";
+import { HomeAssistant } from "../../../types";
+import { hasConfigOrEntityChanged } from "../common/has-changed";
+import "../components/hui-generic-entity-row";
+import { createEntityNotFoundWarning } from "../components/hui-warning";
+import { EntityConfig, LovelaceRow } from "./types";
+
+@customElement("hui-humidifier-entity-row")
+class HuiHumidifierEntityRow extends LitElement implements LovelaceRow {
+  @property() public hass?: HomeAssistant;
+
+  @property() private _config?: EntityConfig;
+
+  public setConfig(config: EntityConfig): void {
+    if (!config || !config.entity) {
+      throw new Error("Invalid Configuration: 'entity' required");
+    }
+
+    this._config = config;
+  }
+
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    return hasConfigOrEntityChanged(this, changedProps);
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this._config) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+
+    if (!stateObj) {
+      return html`
+        <hui-warning>
+          ${createEntityNotFoundWarning(this.hass, this._config.entity)}
+        </hui-warning>
+      `;
+    }
+
+    return html`
+      <hui-generic-entity-row .hass=${this.hass} .config=${this._config}>
+        <ha-entity-humidifier
+          .hass=${this.hass}
+          .stateObj=${stateObj}
+        ></ha-entity-humidifier>
+      </hui-generic-entity-row>
```

Comment:
```suggestion
      <hui-generic-entity-row .hass=${this.hass} .config=${this._config} .secondaryText=${"Current Humidity: 28%"}>
        <ha-entity-humidifier
          .hass=${this.hass}
          .stateObj=${stateObj}
        ></ha-entity-humidifier>
      </hui-generic-entity-row>
```

Reply:
And then you also wont need the other file. Just insert the toggle into the generic row

Reply:
Ok, will do that.

Reply:
Done, please check

![humidifier_comparison](https://user-images.githubusercontent.com/2741408/93939354-5afe5c00-fd33-11ea-98d8-7fd7d08e2777.png)